app/scripts/modules/core/src/managed/versionsHistory/VersionContent.tsx

```diff
@@ -14,7 +20,18 @@ interface IVersionInEnvironmentProps {
   envPinnedVersions?: PinnedVersions[keyof PinnedVersions];
 }
 
+const useGetDetailedVersionData = ({ environment, version }: Omit<IVersionInEnvironmentProps, 'envPinnedVersions'>) => {
+  const app = useApplicationContextSafe();
+  const { data: detailedVersionData, error, loading } = useFetchVersionQuery({
+    variables: { appName: app.name, versions: [version.version] },
+  });
+  const environmentData = detailedVersionData?.application?.environments?.find((env) => env.name === environment);
+  const artifactData = environmentData?.state.artifacts?.find((artifact) => artifact.reference === version.reference);
+  return { detailedVersionData: artifactData?.versions?.find((v) => v.version === version.version), error, loading };
```

Comment:
There are lots of optional fields in these interfaces.  Can you comment about why that is?  Is this an artifact of the gql schema?

Reply:
Yes, in general, it's recommended to use nullable fields in graphql except for identifier, it makes your JS code a bit though... 
I'll share with you a doc with some context